src/pages/workspaces/workspace/LockWorkspaceModal.js

```diff
@@ -0,0 +1,41 @@
+import { useState } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { ButtonPrimary, spinnerOverlay } from 'src/components/common'
+import Modal from 'src/components/Modal'
+import { Ajax } from 'src/libs/ajax'
+import { reportError } from 'src/libs/error'
+
+
+const LockWorkspaceModal = ({ workspace: { workspace: { namespace, name, isLocked } }, onDismiss, onSuccess }) => {
+  const [togglingLock, setTogglingLock] = useState(false)
+  const helpText = isLocked ? 'Unlock Workspace' : 'Lock Workspace'
+
+  const toggleWorkspaceLock = async () => {
+    try {
+      setTogglingLock(true)
+      isLocked ? await Ajax().Workspaces.workspace(namespace, name).unlock() : await Ajax().Workspaces.workspace(namespace, name).lock()
+      onDismiss()
+      onSuccess()
+    } catch (error) {
+      reportError('Error toggling workspace lock', error)
+      setTogglingLock(false)
+    }
+  }
+  return h(Modal, {
+    title: helpText,
+    onDismiss,
+    okButton: h(ButtonPrimary, {
+      onClick: toggleWorkspaceLock,
+      tooltip: helpText
```

Comment:
Should this button have a tooltip if the tooltip text is the same as the button text?